src/main/java/ui/UI.java

```diff
@@ -443,6 +448,27 @@ private void primaryRepoChanged(String repoId) {
         logic.openPrimaryRepository(repoId);
         logic.setDefaultRepo(repoId);
     }
+    
+    public void switchDefaultRepo(){
+        logger.info("Triggered switch default repo");
+        String[] openRepos = logic.getOpenRepositories().toArray(new String[0]);
```

Comment:
`openRepos` might not be sorted as the sorting is done in RepositorySelector. Why not just get the contents from the RepositorySelector's combo box?